code/game/gamemodes/events/space_weather.dm

```diff
@@ -0,0 +1,295 @@
+
+/datum/event/harmonic_feedback/end()
+	command_announcement.Announce("The gravitational wavefronts have passed.", "Harmonic Feedback Surge Anomaly")
+
+/datum/event/harmonic_feedback/tick() //around two seconds
+	for(var/obj/machinery/power/shield_generator/G in GLOB.machines)
+		G.take_damage(10, SHIELD_DAMTYPE_EM)
+
+/datum/storyevent/micro_debris
+	id = "micro_debris"
+	name = "micro debris field"
+	weight = 0.9
+	parallel = FALSE
+	event_type = /datum/event/micro_debris
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_DESTRUCTIVE, TAG_NEGATIVE)
+
+/datum/event/micro_debris
+	startWhen	= 1
+	announceWhen = 2
+	endWhen		= 60
+	var/list/debris_types = list(
+		/obj/effect/meteor/dust/glass=40,\
+		/obj/effect/meteor/dust/rods=30,\
+		/obj/effect/meteor/dust/metal=20,\
+		/obj/effect/meteor/dust/ice=10
+	)
+
+/datum/event/micro_debris/announce()
+	command_announcement.Announce("The ship is now passing through a micro debris field.", "Micro Debris Field Alert")
+
+/datum/event/micro_debris/end()
+	command_announcement.Announce("The ship has now passed through the micro debris field.", "Micro Debris Field Notice")
+
+/datum/event/micro_debris/tick()
+	if(prob(30))
+		for(var/i in 0 to rand(1,3))
+			spawn_debris(pickweight(debris_types), pick(cardinal), pick(GLOB.maps_data.station_levels))
+
+/datum/event/micro_debris/proc/spawn_debris(debris, start_side, zlevel)
+	var/turf/start_turf = spaceDebrisStartLoc(start_side, zlevel)
+	var/turf/destination = spaceDebrisFinishLoc(start_side, zlevel)
+	var/obj/effect/meteor/M = new debris(start_turf)
+	M.dest = destination
+	walk_towards(M, M.dest, 1)
+	return
+
+/datum/storyevent/graveyard
+	id = "graveyard"
+	name = "Space Graveryard"
+	weight = 0.6
+	parallel = FALSE
+	event_type = /datum/event/graveyard
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/graveyard
+	startWhen = 1
+	announceWhen = 3
+	endWhen = 350
+
+/datum/event/graveyard/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/graveyard/start()
+	GLOB.GLOBAL_SANITY_MOD = 1.5
+
+/datum/event/graveyard/announce()
+	command_announcement.Announce("Drifting wrecks of a space station have been detected near the ship. Telecommunication systems are not responsible for any strain on the crew's psychological wellbeing.", "Space Graveyard")
+
+/datum/event/graveyard/tick()
+	if(prob(5)) //random broadcasts
+		var/message = pick("They are ", "He is ", "All of them are ", "I'm ", "We are ")
+		message += pick("going to die... ", "about to turn into those spider mutants... ", "being forcefully converted... ")
+		message += pick("Run while you still can.", "Help!", "Angels bless our souls...", "It's... too late.")
+
+		global_announcer.autosay(message, "Emergency Broadcast")
+	else if(prob(5)) //predetermined broadcasts
```

Comment:
this needs alot more variety in the messages, and to have them fire off slower. we heard the same 6 or so messages repeatedly

Reply:
Proposal to generate those procedurally is still on the table.